tensorwaves/interfaces.py

```diff
@@ -0,0 +1,102 @@
+"""Defines top level interfaces of tensorwaves.
+
+"""
+
+from abc import ABC, abstractmethod
+
+
+class Function(ABC):
```

Comment:
Sufficient for treating intensities? Or should there be a derived intensity class or so?

Reply:
yeah `Intensity` would be just a special type of `Function` (regarding the return value)